scripts/partition_manager/partition_manager.rst

```diff
@@ -54,11 +54,14 @@ The Partition Manager configuration can be also provided by a subsystem.
 (Here, *subsystem* means a software component, like support for a file system.)
 Subsystem Partition Manager configurations cannot define image partitions.
 
-To add a Partition Manager configuration file for a subsystem, place the following line in its :file:`CMakeLists.txt` file (this is only required for subsystems, not applications):
+See :file:`subsys/partition_manager/CMakeLists.txt` for details on how to add the subsystem-specific Partition Manager configuration files.
+The code below shows how the ``settings`` subsystem configuration is added.
 
 .. code-block:: cmake
 
-  add_partition_manager_config(pm.yml)
+   if (CONFIG_SETTINGS_FCB OR CONFIG_SETTINGS_NVS)
+     add_partition_manager_config(pm.yml.settings)
+   endif()
```

Comment:
Referring the reader, who might not be an experienced CMake developer, to a CMakeLists.txt file that defines both macros, functions, and uses generator expressions, could be overwhelming, and give a risk he fails to understand where / how he should update this file.

Secondly, just as we dislike patching upstream Zephyr, out users could feel the same regarding patching NCS.

If this is a valid use-case (and I thik it is), could we make this a real and easy to use for NCS downstream developers ?

Reply:
You have a good point. We should move the macro out to `cmake/extensions.cmake` so that out of tree users can use it as well. 

I agree that this is a valid use case. I will create a task for this, so that we can handle it post-release.

Reply:
NCSDK-5995